modules/client-cache/src/test/scala/higherkindness/mu/rpc/channel/cache/ClientCacheTests.scala

```diff
@@ -65,30 +64,59 @@ class ClientCacheTests extends WordSpec with Matchers {
         cleanUp.millis,
         cleanUp.millis
       )(ConcurrentEffect[IO], cs, timer)
-      _ <- Stream.eval((1 to numClients).toList.traverse_(_ => clientCache.getClient))
+      _ <- Stream.eval((1 to hosts.length).toList.traverse_(_ => clientCache.getClient))
     } yield ()).compile.drain
 
-  def test(numClients: Int, cleanUp: Int): (Int, Int) =
+  def test1(numClients: Int, cleanUp: Int): (Int, Int) =
     (for {
       ref1         <- Ref.of[IO, Int](0)
       ref2         <- Ref.of[IO, Int](0)
-      _            <- compiledStream(ref1, ref2, numClients, cleanUp)
+      _            <- compiledStream(ref1, ref2, (1 to numClients).toList.map(i => ("localhost", i)), cleanUp)
       numCreations <- ref1.get
       numCloses    <- ref2.get
     } yield (numCreations, numCloses)).unsafeRunSync()
 
-  "ClientCache.impl" should {
+  def test2(numClients: Int, cleanUp: Int): (Int, Int) =
```

Comment:
I think from the `ClientCache` perspective, it doesn't matter the type of the key and since this is simpler (just a string) I'll remove the previous test and keep this one